src/stdune/path.ml

```diff
@@ -1194,7 +1196,8 @@ let touch p =
     | In_build_dir k ->
       Kind.to_string (Kind.append_local (Fdecl.get Build.build_dir) k)
   in
-  Unix.utimes p 0.0 0.0
+  try Unix.utimes p 0.0 0.0
+  with Unix.Unix_error (Unix.ENOENT, _, _) -> close_out (open_out p)
```

Comment:
It's better to use `Unix.openfile`/`Unix.close` here, as `open_out` will allocate a big buffer for nothing